src/EnergyPlus/Pumps.cc

```diff
@@ -2051,7 +2051,7 @@ void CalcPumps(EnergyPlusData &state, int const PumpNum, Real64 const FlowReques
                             state.dataPumps->PumpEquip(PumpNum).PartLoadCoef(4) * pow_3(PartLoadRatio);
         state.dataPumps->Power = FracFullLoadPower * state.dataPumps->PumpEquip(PumpNum).NomPowerUse;
 
-    } else if (PumpType == iPumpType::Bank_ConSpeed || PumpType == iPumpType::Bank_VarSpeed) {
+    } else if BITF_TEST_ANY(BITF(PumpType), BITF(iPumpType::Bank_ConSpeed) | BITF(iPumpType::Bank_VarSpeed)) {
 
```

Comment:
If you have an if-else ladder of these things, you can also use switch/case:

```
switch(PumpType) {
   case iPumpType::ConSpeed:
   case iPumpType::VarSpeed:
   case iPumpType::Cond: 
        blahblahblah;
        break;
   case iPumpType::Bank_ConSpeed:
   case iPumpType::Bank_VarSpeed:
       yadayadayadah;
       break;
   default:
      thisisprobablyanerror;
      break;
}
```

Reply:
Thank you. Is there a preferred/optimum way to do it (Or decide between if-else/switch/BITF methods)? In this case, I am guessing BITF doesn't give many advantages since there are just three+two items being compared. So it is best to change to switch/case with the most common cases put first?

Reply:
There are actually two separate issues here. 

Issue 1. Should you use `Switch/case` or `if-else`? You should use `switch/case` if the `if-else` ladder has at least three or four cases and one of them is not dynamically dominant. There is also the limitation that `switch/case` is also for single variable equality tests, you cannot have a compound conditional in a `case`. That is the specific case here. For all other situations you either have to use `if-else` or you don't have to, but you should. This brings us to issue 2.

Issue 2. If you are using `if-else`, should you use `BITF`? The test here is how many things you are comparing against. `BITF` is three instructions including one branch for any number of comparisons. `X == A || X == B ...` is two instructions including one branch _per_ comparison. By the numbers (especially the branch numbers) you want to use `BITF` anytime you are comparing against two or more things. The performance advantage for two comparisons is admitted small, but it is there. One disclaimer is that this only true if the things you are comparing against are enumerations or other small compile-time constants.

Reply:
Thank you very much again. Last question: does dynamically dominant mean that the probability of a case occurring is higher than others?

Reply:
That's right. If one of the cases occurs 90% of the time, you want to use an `if-then` ladder and put that case first. The remaining cases will be a bit slower than they would have been had you used a `switch\case`, but this case will be faster and that is the one that matters for performance.

Reply:
Got it. Thanks so much!